src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,51 @@
+import { Component, type ReactNode } from 'react';
+import '../styles/ErrorBoundary.scss';
+
+interface Props {
+  children: ReactNode;
+}
+
+interface State {
+  message: string | null;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  timeoutId: number | null = null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { message: null };
+  }
```

Comment:
instead of a constructor, define it right away
```suggestion
state = { message: null, error: null };
```